services/refCompiler.js

```diff
@@ -0,0 +1,154 @@
+var fs = require('fs');
```

Comment:
It is not the same. Not sure what the best way is to do a side by side comparison, but at the very least it includes these changes:
- spec files are identified using regular expressions to look for .json and .yaml extensions. This was suggested by @gconsidine because he ran into an issue where swap files were being picked up by ref-compiler during use of vim.
- single compileReferences function was divided into two functions compileYAMLReferences and compileJSONReferences. Both follow similar logical flow but obviously modified to deal with the different data formats.